plugins/cpu/cpu.go

```diff
@@ -13,31 +21,170 @@ const (
 	niceGauge
 )
 
+// Hold cpu sampled data and calculate moving averages
+type sample struct {
+	alpha        float64
+	values       []float64
+	expMovingAvg float64
+}
+
+func newSample(alpha float64) *sample {
+	return &sample{alpha: alpha}
+}
+
+func (s *sample) add(v float64) {
+	s.values = append(s.values, v)
+
+	if s.expMovingAvg == 0 {
+		s.expMovingAvg = v
```

Comment:
This is the initialization case if I'm not mistaken, should be done only when len(s.values) was 0 on entry to the func? Otherwise, if the exp moving avg ever got back to 0 we'd re-init.